lib/_stream_transform.js

```diff
@@ -133,89 +99,67 @@ function Transform(options) {
       this._flush = options.flush;
   }
 
-  // When the writable side finishes, then flush out anything remaining.
+  // TODO(ronag): Unfortunately _final is invoked asynchronously.
+  // Use `prefinish` hack. `prefinish` is emitted synchronously when
+  // and only when `_final` is not defined. Implementing `_final`
+  // to a Transform should be an error.
```

Comment:
I would recommend to open an issue about this. `_final` _should_ be safe in both `Readable` and `Transform`. Good spot.

Reply:
https://github.com/nodejs/node/pull/32780